### PR TITLE
NFC: Update `replaceMemrefUsesAndPropagateType` to use a more general transitive replacement.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
The current utility method only looked at subviews and replaced uses there. This is not general enough. #12510 recently added a more general method to replace modified `memref` values transitively. Use this method as the implementation of the
`replaceMemrefUsesAndPropagateType` method and common out all the uses.

### CODE REVIEW

**Comment by MaheshRavishankar** [comment]
#12617 has a use to common out here as well. Will wait for that to land before landing this

**Review by antiagainst — APPROVED** [review]

**Pull request merged**
